src/ccf/calc_ccf.jl

```diff
@@ -281,6 +281,38 @@ function ccf_1D(λ::A2, flux::A3, #line_list::ALL, #mask_shape1::A3
     return ccf_out
 end
 
+"""
+    `ccf_1D( λs, fluxes; ccf_plan )`
+    Compute the cross correlation functions of spectra with a mask.
+# Inputs:
+- `λs`: 1-d array of wavelengths
+- `fluxes`:  2-d array of fluxes, individual spectra along first dim
+# Optional Arguments:
+- `ccf_plan`:  parameters for computing ccf (BasicCCFPlan())
+# Returns:
+- 2-d array of size (length(calc_ccf_v_grid(plan)), size(flux, 2))
+"""
+function ccf_1D(λ::A2, flux::A3, #line_list::ALL, #mask_shape1::A3
+                plan::PlanT = BasicCCFPlan() ) where {
+                #; mask_shape::ACMS = TopHatCCFMask(), plan::PlanT = BasicCCFPlan() ) where {
+                T1<:Real, A1<:AbstractArray{T1,1}, T2<:Real, A2<:AbstractArray{T2,1}, T3<:Real, A3<:AbstractArray{T3,2},
+                #ALL<:AbstractLineList, ACMS<:AbstractCCFMaskShape, AP<:AbstractCCFPlan }
+                PlanT<:AbstractCCFPlan } # ALL<:AbstractLineList, ACMS<:AbstractCCFMaskShape }
+    @assert ndims(λ) == 1
+    @assert ndims(flux) == 2
+    @assert length(λ) == size(flux, 1)
+    len_v_grid = calc_length_ccf_v_grid(plan)
+    ccf_out = zeros(len_v_grid, size(flux, 2))
+    if length(plan.line_list) < 1  # If no lines in chunk
+        return ccf_out
+    end
+
+    for i in 1:size(flux, 2)
+        ccf_out[:,i] = ccf_1D(λ, flux[:,i], plan)
```

Comment:
I'd suggest using ccf_1D!(ccf_out, λs, fluxes; ccf_plan ) here

Reply:
I think I made a somewhat kluge-y implementation (see below). Passing a slice of ccf_out (i.e., ccf_out[:,i]) seems to edit a copy of the slice, not the slice itself.    

```julia
    ccf_temp = zeros(len_v_grid)
    for i in 1:size(flux, 2)
        ccf_1D!(ccf_temp, λ, flux[:,i], plan)
        ccf_out[:,i] = ccf_temp
    end
```

Reply:
Try using a view
```julia
    for i in 1:size(flux, 2)
        ccf_1D!(view(ccf_out, 1:len_v_grid, i), λ, flux[:,i], plan)
    end
```

Reply:
Even better, use a view for flux, too

Reply:
Done. I just used a colon instead of ```1:len_v_grid```.